Sources/MapboxCoreNavigation/NavigationEventsManager.swift

```diff
@@ -14,6 +27,32 @@ public protocol EventsManagerDataSource: AnyObject {
     var locationProvider: NavigationLocationManager.Type { get }
 }
 
+open class FreeDriveNavigationEventsManager: NavigationEventsManager {
```

Comment:
Would it be possible to genericize NavigationEventsManager to avoid the need for subclassing? I think it could become problematic that NavigationEventsManager is being treated almost as an abstract base class, due to these overrides, even though it normally isn’t abstract. It’s reminiscent of how NavigationMapView used to subclass MGLMapView but had to work around increasingly complex logic specific to non-navigation maps.

If separate types are necessary for active navigation versus passive navigation, then I’d suggest factoring out whatever is in common between both activities into a new base class or protocol with default implementations.

Reply:
I agree, removed the inheritance here. NavigationEventsManager now has two optional data sources – one for passive navigation, one for active. The idea of creating a data source protocol doesn't work in this case, the reported values are absolutely different.